plugins/outputs/arc/arc.go

```diff
@@ -0,0 +1,342 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package arc
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	_ "embed"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const (
+	defaultURL         = "http://localhost:8000/api/v1/write/msgpack"
+	defaultTimeout     = 5 * time.Second
+	defaultContentType = "application/msgpack"
+	defaultUserAgent   = "Telegraf-Arc-Output-Plugin"
+	defaultBatchSize   = 1000
+)
+
+// Arc output plugin for writing metrics to Arc time-series database using MessagePack binary protocol
+type Arc struct {
+	// Arc MessagePack API URL
+	URL string `toml:"url"`
+
+	// HTTP timeout
+	Timeout config.Duration `toml:"timeout"`
+
+	// API Key for authentication
+	APIKey config.Secret `toml:"api_key"`
+
+	// Database name for multi-database architecture
+	// Routes metrics to a specific database namespace (e.g., "production", "staging", "default")
+	Database string `toml:"database"`
+
+	// HTTP Headers
+	Headers map[string]string `toml:"headers"`
+
+	// Content encoding: "gzip" or "identity" (default: gzip)
+	ContentEncoding string `toml:"content_encoding"`
+
+	// User agent string
+	UserAgent string `toml:"user_agent"`
+
+	// Batch size for MessagePack writes
+	BatchSize int `toml:"batch_size"`
+
+	// Log
+	Log telegraf.Logger `toml:"-"`
+
+	client *http.Client
+}
+
+// ArcColumnarData represents columnar format data for Arc's MessagePack format
+type ArcColumnarData struct {
+	Measurement string                 `msgpack:"m"`
+	Columns     map[string]interface{} `msgpack:"columns"`
+}
+
+func (*Arc) SampleConfig() string {
+	return sampleConfig
+}
+
+func (a *Arc) Init() error {
+	// Set defaults
+	if a.URL == "" {
+		a.URL = defaultURL
+	}
+
+	if a.Timeout == 0 {
+		a.Timeout = config.Duration(defaultTimeout)
+	}
+
+	if a.UserAgent == "" {
+		a.UserAgent = defaultUserAgent
+	}
+
+	if a.ContentEncoding == "" {
+		a.ContentEncoding = "gzip"
+	}
+
+	if a.BatchSize == 0 {
+		a.BatchSize = defaultBatchSize
+	}
+
+	return nil
+}
+
+func (a *Arc) Connect() error {
+	a.client = &http.Client{
+		Timeout: time.Duration(a.Timeout),
+		Transport: &http.Transport{
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 20,
+			IdleConnTimeout:     90 * time.Second,
```

Comment:
I don't think hardcoding this is a good idea! Use the [`HTTPClientConfig`](https://github.com/influxdata/telegraf/blob/b69f25a9a47b4ca1adf48904cfe9ae5cb5aef9e1/plugins/common/http/config.go#L53) (mentioned above) instead and get those options for free.